### PR TITLE
[IMPROVE] Migrate UIKit/MultiSelect to ActionSheet

### PR DESCRIPTION
**Pull request by try-catch-stack**
## Proposed changes

* `MultiSelect` component of the UIKit was implemented using `Animated` logic from React Native. This PR migrates the component to render as ActionSheet.

## Screenshots

https://user-images.githubusercontent.com/77742477/171036477-061fff7f-8c00-4634-8371-4cb3723849dc.mp4

|Before | After|
|--------|------|
|Screenshot_1653930643|Screenshot_1654019811|
|Screenshot_1653930615|Screenshot_1654019787|
|Screenshot_1653930607|Screenshot_1654020330|

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] Improvement (non-breaking change which improves a current function)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation update (if none of the other choices apply)

## Checklist

- [X] I have read the [CONTRIBUTING](https://github.com/RocketChat/Rocket.Chat/blob/develop/.github/CONTRIBUTING.md#contributing-to-rocketchat) doc
- [X] I have signed the [CLA](https://cla-assistant.io/RocketChat/Rocket.Chat.ReactNative)
- [X] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if applicable)
- [ ] I have added necessary documentation (if applicable)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/containers/ActionSheet/ActionSheet.tsx [hunk 1]
```diff
@@ -96,7 +96,7 @@ const ActionSheet = React.memo(
 
 		const renderHandle = () => (
 			<>
-				<Handle />
+				<Handle backgroundColor={data?.backgroundColor ? data.backgroundColor : colors.focusedBackground} />
```
Comment (diegolmello) [inline app/containers/ActionSheet/ActionSheet.tsx:99]: I think we shouldn't give the option to send other colors here.
We should unify the behavior instead, using the same color everywhere.
You can keep the previous color the default one.

File: app/containers/UIKit/MultiSelect/MultiSelectContent.tsx [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+		theme,
+		multiselect,
+		select,
+		onChange,
+		setCurrentValue,
+		onHide,
+		selectedItems
+	}: IMultiSelectContentProps) => {
+		const { colors } = useTheme();
+		const [selected, setSelected] = useState<any>(Array.isArray(selectedItems) ? selectedItems : []);
```
Comment (diegolmello) [inline app/containers/UIKit/MultiSelect/MultiSelectContent.tsx:37]: We need to properly type this

File: app/containers/UIKit/MultiSelect/MultiSelectContent.tsx [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+interface IMultiSelectContentProps {
+	onSearch?: () => void;
+	options?: IItemData[];
+	theme: TSupportedThemes;
+	multiselect: boolean;
+	select: React.Dispatch<any>;
+	onChange: Function;
+	setCurrentValue: React.Dispatch<React.SetStateAction<string>>;
+	onHide: Function;
+	selectedItems: any;
```
Comment (diegolmello) [inline app/containers/UIKit/MultiSelect/MultiSelectContent.tsx:21]: Missing definition

File: app/containers/UIKit/MultiSelect/index.tsx [hunk 4]
```diff
@@ -71,43 +47,46 @@ export const MultiSelect = React.memo(
...
+						theme={theme}
+						select={select}
+						onChange={onChange}
+						setCurrentValue={setCurrentValue}
+						onHide={onHide}
+						multiselect={multiselect}
+						selectedItems={selected}
+					/>
+				),
+				snaps: [300],
```
Comment (diegolmello) [inline app/containers/UIKit/MultiSelect/index.tsx:83]: I'm not sure we need this

File: app/containers/UIKit/MultiSelect/index.tsx [hunk 5]
```diff
@@ -71,43 +47,46 @@ export const MultiSelect = React.memo(
 		inputStyle,
 		innerInputStyle
 	}: IMultiSelect) => {
-		const { theme } = useTheme();
+		const { theme, colors } = useTheme();
```
Comment (diegolmello) [inline app/containers/UIKit/MultiSelect/index.tsx:50]: Is it too cumbersome to migrate to colors on this PR? (dropping theme)

File: app/containers/UIKit/MultiSelect/MultiSelectContent.tsx [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+				setSelected(newSelect);
+				select(newSelect);
+			} else {
+				onChange({ value });
+				setCurrentValue(text);
+				onHide();
+			}
+		};
+
+		const items: any = onSearch
```
Comment (diegolmello) [inline app/containers/UIKit/MultiSelect/MultiSelectContent.tsx:61]: type it

File: app/containers/UIKit/MultiSelect/MultiSelectContent.tsx [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+			? options
+			: options?.filter((option: any) => textParser([option.text]).toLowerCase().includes(search.toLowerCase()));
+
+		return (
+			<View style={[styles.actionSheetContainer, { backgroundColor: colors.backgroundColor }]}>
+				<FormTextInput
+					testID='multi-select-search'
+					onChangeText={onSearch || onSearchChange}
```
Comment (diegolmello) [inline app/containers/UIKit/MultiSelect/MultiSelectContent.tsx:69]: this is awkward

**Review by diegolmello — CHANGES_REQUESTED** [review]

File: app/containers/UIKit/MultiSelect/index.tsx (hunk 4 above)
Comment (try-catch-stack) [reply app/containers/UIKit/MultiSelect/index.tsx]: We do need to specify the snaps here as we're using the custom component logic but we'll surely want multiple snaps for longer FlatLists. I did this just to match the previous design and will update the logic here.

File: app/containers/UIKit/MultiSelect/MultiSelectContent.tsx (hunk 7 above)
Comment (try-catch-stack) [reply app/containers/UIKit/MultiSelect/MultiSelectContent.tsx:58]: onChangeText?

**Comment by GleidsonDaniel** [comment]
@try-catch-stack can you fix the conflicts?

**Comment by try-catch-stack** [comment]
Done @GleidsonDaniel

**Pull request merged**